src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -532,92 +560,106 @@ namespace OpenMS
     OpenSwath::Scoring::normalize_sum(&normalized_library_intensity[0], boost::numeric_cast<int>(normalized_library_intensity.size()));
   }
 
-  OpenSwath::SpectrumPtr OpenSwathScoring::fetchSpectrumSwath(OpenSwath::SpectrumAccessPtr swath_map,
-                                                              double RT, int nr_spectra_to_add, const double drift_lower, const double drift_upper)
+  std::vector<OpenSwath::SpectrumPtr> OpenSwathScoring::fetchSpectrumSwath(OpenSwath::SpectrumAccessPtr swathmap, double RT, int nr_spectra_to_add, double drift_lower, double drift_upper)
   {
-    return getAddedSpectra_(std::move(swath_map), RT, nr_spectra_to_add, drift_lower, drift_upper);
+
+    std::vector<OpenSwath::SpectrumPtr> all_spectra = fetchMultipleSpectra_(swathmap, RT, nr_spectra_to_add);
+    if (spectra_addition_method_ == "simple")
```

Comment:
only a comment, since not part of this PR: should ideally be an `enum class`, not a string (easy to get the value wrong).

Reply:
the comparison will also be a tad faster